go/vt/vtctl/vtctl.go

```diff
@@ -2132,34 +2136,100 @@ func commandApplyVSchema(ctx context.Context, wr *wrangler.Wrangler, subFlags *f
 	if subFlags.NArg() != 1 {
 		return fmt.Errorf("the <keyspace> argument is required for the ApplyVSchema command")
 	}
-	if (*vschema == "") == (*vschemaFile == "") {
-		return fmt.Errorf("either the vschema or vschemaFile flag must be specified when calling the ApplyVSchema command")
-	}
-	var schema []byte
-	if *vschemaFile != "" {
-		var err error
-		schema, err = ioutil.ReadFile(*vschemaFile)
+	keyspace := subFlags.Arg(0)
+
+	var vs *vschemapb.Keyspace
+	var err error
+	if *sql != "" || *sqlFile != "" {
```

Comment:
TIOLI. big if/else blocks make my eyes cross:

```
	// apparently golang doesn't have an xor operator
	ddlMode := (*sql != "") != (*sqlFile != "")
	vschemaMode := (*vschema != "") != (*vschemaFile != "")

	switch {
	case ddlMode && vschemaMode:
		// both specified. return err
	case !ddlMode && !vschemaMode:
		// neither specified, return err
	}

	if ddlMode {

	}

	if vschemaMode {

	}
```

Reply:
good suggestion. changed the names a bit but this is a much more readable cleanup